src/extensibility/ExtensionManagerView.js

```diff
@@ -336,6 +337,19 @@ define(function (require, exports, module) {
             context[helper] = registry_utils[helper];
         });
 
+        // Do some extra validation on homepage url to make sure we don't end up executing local binary
+        if (context.metadata.homepage) {
+            var parsed = PathUtils.parseUrl(context.metadata.homepage);
+            // Check if the homepage refers to a local resource
+            if (parsed.protocol.trim().toLowerCase() === "file:") {
```

Comment:
@swmitra If have something like `C:\Windows\System32\notepad.exe`, will `parsed.protocol` list it under `file:`?